sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -160,8 +160,7 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
       // We can't reuse the broadcast because the join type doesn't support broadcast,
       // and doing DPP means running an extra query that may have significant overhead.
       // We need to make sure the pruning side is very big so that DPP is still worthy.
-      canBroadcastBySize(otherPlan, conf) &&
```

Comment:
Previously, we only check `canBroadcastBySize` in the else branch. That means, if broadcastOnly=false, we can do DPP if the join type allows doing broadcast, and we don't care if the size is large or not, we only care the cost `estimatePruningSideSize > overhead`.

This is a bit aggressive but it was the behavior since the beginning. Shall we restore the behavior? @wangyum

Reply:
Do you mean replace `estimatePruningSideSize * conf.dynamicPartitionPruningPruningSideExtraFilterRatio > overhead` to `estimatePruningSideSize > overhead`?